hr_timesheet_hours_month/models/project_task.py

```diff
@@ -0,0 +1,34 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import fields, models, _
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    planned_hours_year = fields.Boolean(default=False)
+
+    remaining_hours_month = fields.Float(compute='_compute_remaining_hours_month', compute_sudo=True)
+
+    def _compute_remaining_hours_month(self):
+        n_month = fields.Date.today().month
+        for record in self:
+            if record.planned_hours_year:
+                record.remaining_hours_month = record.remaining_hours - (record.planned_hours - (n_month * (record.planned_hours / 12)))
+            else:
+                record.remaining_hours_month = 0.0
+    
+    def name_get(self):
+        name_mapping = dict(super().name_get())
```

Comment:
In parent code (in `hr_timesheet`) remaining hours text is only added when these conditions are met:
- `hr_timesheet_display_remaining_hours` is provided by context.
- task is allowed for timesheets, and
- planned hours are positive.

Only the last one is covered, we should add the other restrictions.